[std] Remove problematic phrases from notes (#6758)
cplusplus/draft: source/overloading.tex

or when converting to the left operand of an assignment operation
only standard conversion sequences are allowed.
\begin{note}

> A conversion to the explicit object parameter, if any,
> can include user-defined conversion sequences.

Reviewer: ```suggestion
When a conversion to the explicit object parameter occurs,
it can include user-defined conversion sequences.
```

?

Author: Very nice!
